### PR TITLE
Full tvOS support

### PR DESCRIPTION
**Pull request by addb**
## Screenshots (if appropriate):
AppleTV app

First screen: 
Simulator Screen Shot - Apple TV 4K (at 1080p) (2nd generation) - 2022-06-28 at 16 37 10

Video player:
Simulator Screen Shot - Apple TV 4K (at 1080p) (2nd generation) - 2022-06-28 at 16 37 26

Version in pings:
iOS:
Screen Shot 2022-06-28 at 5 25 22 PM

tvOS:
Screen Shot 2022-06-28 at 5 25 34 PM

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: Script/test-podspec.sh [hunk 1]
```diff
@@ -44,3 +44,44 @@ xcodebuild clean build -scheme TestProject-Package -workspace TestProject.xcwork
...
+echo '############# Build for generic tvOS device ###############'
+xcodebuild build -scheme TestProject-Package -workspace TestProject.xcworkspace -destination 'generic/platform=tvOS'
+
+# Archive for generic tvOS device
+echo '############# Archive for generic tvOS device ###############'
+xcodebuild archive -scheme TestProject-Package -workspace TestProject.xcworkspace -destination 'generic/platform=tvOS Simulator'
+
+# Build for generic tvOS simulator
+echo '############# Build for x86_64 simulator ###############'
+xcodebuild build -scheme TestProject-Package -workspace TestProject.xcworkspace -destination 'generic/platform=tvOS Simulator'
```
Comment (emdobrin) [inline Script/test-podspec.sh:83]: tvOS simulator

File: AEPMedia.xcodeproj/project.pbxproj [hunk 2]
```diff
@@ -1137,8 +1296,11 @@
 				MTL_ENABLE_DEBUG_INFO = NO;
 				MTL_FAST_MATH = YES;
 				SDKROOT = iphoneos;
+				SUPPORTED_PLATFORMS = "iphonesimulator iphoneos appletvos appletvsimulator";
```
Comment (emdobrin) [inline AEPMedia.xcodeproj/project.pbxproj:1299]: do we need to have SDKROOT defined for lib targets? some of our other repos don't seem to have it (same for the debug target above)

Comment (addb) [reply AEPMedia.xcodeproj/project.pbxproj:1298]: yes we have Base SDK configured for all

File: .circleci/config.yml [hunk 3]
```diff
@@ -49,33 +49,46 @@ jobs:
...
+          name: Pre-start tvOS Simulator
+          command: xcrun instruments -w "Apple TV (15.0) [" || true
+
+      - run:
+          name: Run tvOS Tests
+          command: make test-tvos
+
+      - run:
+          name: Upload Code Coverage Reports for tvOS
+          command: bash ./codecov -v -X s3 -c -D "./build/out" -J "AEPCore tvOS"
```
Comment (emdobrin) [inline .circleci/config.yml:85]: -J "AEPMedia tvOS"

File: AEPMedia.podspec [hunk 4]
```diff
@@ -11,6 +11,7 @@ The Media library provides APIs that allow use of the Media Analytics product in
   s.source           = { :git => "https://github.com/adobe/aepsdk-media-ios", :tag => s.version.to_s }
 
   s.ios.deployment_target = '10.0'
+  s.tvos.deployment_target = '10.0'
 
   s.swift_version = '5.1'
```
Comment (emdobrin) [inline AEPMedia.podspec:17]: we should update min core 3.7 too

File: AEPMedia.xcodeproj/xcshareddata/xcschemes/AEPMediaFunctionalTests.xcscheme [hunk 5]
```diff
@@ -5,6 +5,22 @@
    <BuildAction
       parallelizeBuildables = "YES"
       buildImplicitDependencies = "YES">
+      <BuildActionEntries>
```
Comment (emdobrin) [inline AEPMedia.xcodeproj/xcshareddata/xcschemes/AEPMediaFunctionalTests.xcscheme:8]: can we get the scheme and xcscheme file renamed as well to FunctionalTests

File: AEPMedia.xcodeproj/xcshareddata/xcschemes/AEPMediaUnitTests.xcscheme [hunk 6]
```diff
@@ -5,20 +5,47 @@
    <BuildAction
       parallelizeBuildables = "YES"
       buildImplicitDependencies = "YES">
+      <BuildActionEntries>
```
Comment (emdobrin) [inline AEPMedia.xcodeproj/xcshareddata/xcschemes/AEPMediaUnitTests.xcscheme:8]: should rename to UnitTests

File: codecov.yml [hunk 7]
```diff
@@ -26,6 +26,7 @@ coverage:
     - "./Documentation"
     - "./Script"
     - "./Samples"
```
Comment (emdobrin) [inline codecov.yml:28]: we can remove Samples now

**Review by emdobrin — APPROVED** [review]

**Pull request merged**